### PR TITLE
[CIRCLE-29501] Add commands to manage runner types

### PR DESCRIPTION
**Pull request by pete-woods**
Commands added:
- `circleci runner resource-class create <resource-class> <description>`
- `circleci runner resource-class delete <resource-class>`
- `circleci runner resource-class list <namespace>`
- `circleci runner token create [--config] <resource-class> <nickname>`
- `circleci runner token delete <token-id>`
- `circleci runner token list <resource-class>`

Aliases of `delete` to `rm` and `list` to `ls` have been added.

### CODE REVIEW

File: api/runner/runner.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+	resp := struct {
+		Items []ResourceClass `json:"items"`
+	}{}
+	_, err = r.rc.DoRequest(req, &resp)
+	return resp.Items, err
+}
+
+func (r *Runner) DeleteResourceClass(id string) error {
+	req, err := r.rc.NewRequest("DELETE", &url.URL{Path: fmt.Sprintf("runner/resource/%s", id)}, nil)
```
Comment (glenjamin) [inline api/runner/runner.go:61]: `id` should be URL escaped

File: api/runner/runner.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+type Token struct {
+	ID            string    `json:"id"`
+	Token         string    `json:"token"`
+	ResourceClass string    `json:"resource_class"`
+	Nickname      string    `json:"nickname"`
+	CreatedAt     time.Time `json:"created_at"`
+}
+
+func (t *Token) ToAgentConfig() *AgentConfig {
```
Comment (glenjamin) [inline api/runner/runner.go:78]: I think this should be outside of the `API` package

File: api/runner/runner.go [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+	resp := struct {
+		Items []Token `json:"items"`
+	}{}
+	_, err = r.rc.DoRequest(req, &resp)
+	return resp.Items, err
+}
+
+func (r *Runner) DeleteToken(id string) error {
+	req, err := r.rc.NewRequest("DELETE", &url.URL{Path: fmt.Sprintf("runner/token/%s", id)}, nil)
```
Comment (glenjamin) [inline api/runner/runner.go:125]: `id` should be url-escaped

File: cmd/root_test.go [hunk 4]
```diff
@@ -15,7 +15,7 @@ var _ = Describe("Root", func() {
 	Describe("subcommands", func() {
 		It("can create commands", func() {
 			commands := cmd.MakeCommands()
-			Expect(len(commands.Commands())).To(Equal(17))
+			Expect(len(commands.Commands())).To(Equal(18))
```
Comment (glenjamin) [inline cmd/root_test.go:18]: maybe replace this with a golden test on `--help`?

File: cmd/runner/resource_class.go [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+		Aliases: []string{"ls"},
+		Args:    cobra.ExactArgs(1),
+		PreRunE: preRunE,
+		RunE: func(_ *cobra.Command, args []string) error {
+			rcs, err := r.GetResourceClassesByNamespace(args[0])
+			if err != nil {
+				return err
+			}
+
+			table := tablewriter.NewWriter(os.Stdout)
```
Comment (glenjamin) [inline cmd/runner/resource_class.go:59]: ~ maybe extract and share the table with create?

File: cmd/runner/token.go [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+		Short:   "Create a token for a resource-class",
+		Args:    cobra.ExactArgs(2),
+		PreRunE: preRunE,
+		RunE: func(_ *cobra.Command, args []string) error {
+			token, err := r.CreateToken(args[0], args[1])
+			if err != nil {
+				return err
+			}
+
+			if createOpts.config {
```
Comment (glenjamin) [inline cmd/runner/token.go:33]: This is a great idea!

Maybe it should be on by default?

File: cmd/runner/token.go [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+		Short:   "List tokens for a resource-class",
+		Args:    cobra.ExactArgs(1),
+		PreRunE: preRunE,
+		RunE: func(_ *cobra.Command, args []string) error {
+			tokens, err := r.GetRunnerTokensByResourceClass(args[0])
+			if err != nil {
+				return err
+			}
+
+			table := tablewriter.NewWriter(os.Stdout)
```
Comment (glenjamin) [inline cmd/runner/token.go:70]: ~ same comment as resource class about sharing the table helper

File: cmd/root_test.go (hunk 4 above)
Comment (pete-woods) [reply cmd/root_test.go:18]: Not sure I wanna start re-doing all of the CLI's tests just now!

File: cmd/runner/token.go (hunk 7 above)
Comment (pete-woods) [reply cmd/runner/token.go:70]: Can't do this one, as they are different (one contains the token value, the other does not)

File: cmd/runner/token.go (hunk 6 above)
Comment (pete-woods) [reply cmd/runner/token.go:33]: 🤔

File: cmd/root_test.go (hunk 4 above)
Comment (pete-woods) [reply cmd/root_test.go:18]: I've put in a bunch of usage test assertions for the runner commands.

Once this merges, I'll have a look at how hard it'd be to do the same for all the commands in the CLI.

**Review by danmux — APPROVED** [review]

File: cmd/root_test.go (hunk 4 above)
Comment (glenjamin) [reply cmd/root_test.go:18]: I was thinking just one that lists all of the commands - but I had forgotten that there isn't a help command which lists all commands, and these are hidden anyway!

Comment (pete-woods) [reply cmd/root_test.go:18]: Good point!

**Review by marcomorain — APPROVED** [review]

**Pull request merged**